coderd/database/dbauthz/dbauthz.go

```diff
@@ -1741,6 +1741,18 @@ func (q *querier) GetFileByID(ctx context.Context, id uuid.UUID) (database.File,
 	return file, nil
 }
 
+func (q *querier) GetFileIDByTemplateVersionID(ctx context.Context, templateVersionID uuid.UUID) (uuid.UUID, error) {
+	fileID, err := q.db.GetFileIDByTemplateVersionID(ctx, templateVersionID)
+	if err != nil {
+		return uuid.Nil, err
+	}
+	err = q.authorizeContext(ctx, policy.ActionRead, rbac.ResourceFile.WithID(fileID))
+	if err != nil {
+		return uuid.Nil, err
+	}
```

Comment:
The perm story around files is... not ideal:
https://github.com/coder/coder/issues/6775


I think we should just check if they can `read` the `template_version`. And not check file perms.


See the perm check for updating a file
https://github.com/coder/coder/blob/c5b542a27ee0a5a70d0202a6ff25117ba77049c5/coderd/database/dbauthz/dbauthz.go#L701-L721

Reply:
the `ctx` needs to pass this check anyway, because this is the check that `GetFileById` performs.

so the endpoint is going to need to use `dbauthz.AsProvisionerd` or some other built-in system actor.

Reply:
Yea, we can use a system like user to fetch the files.

We can keep it for now then, can we just throw a comment indicating this is not an ideal check right now?